x-pack/elastic-agent/pkg/artifact/config_test.go

```diff
@@ -0,0 +1,249 @@
+			expectedInstallDirectory: defaultValues.InstallPath,
+			expectedDropDirectory:    defaultValues.DropPath,
+			expectedFingerprint:      "",
+			expectedTLS:              defaultValues.TLS != nil,
+			expectedTLSEnabled:       false,
+			expectedDisableProxy:     defaultValues.Proxy.Disable,
+			expectedTimeout:          defaultValues.Timeout,
+		},
+		{
+			input: `agent.download:
+  source_uri: " "
+  sourceURI: "testing.uri"
+`,
+			initialConfig:            DefaultConfig(),
+			expectedSourceURI:        "testing.uri",
+			expectedTargetDirectory:  defaultValues.TargetDirectory,
+			expectedInstallDirectory: defaultValues.InstallPath,
+			expectedDropDirectory:    defaultValues.DropPath,
+			expectedFingerprint:      "",
+			expectedTLS:              defaultValues.TLS != nil,
+			expectedTLSEnabled:       false,
+			expectedDisableProxy:     defaultValues.Proxy.Disable,
+			expectedTimeout:          defaultValues.Timeout,
+		},
+		{
+			input: `agent.download:
+  source_uri: "testing.uri"
+  sourceURI: " "
+`,
+			initialConfig:            DefaultConfig(),
+			expectedSourceURI:        "testing.uri",
+			expectedTargetDirectory:  defaultValues.TargetDirectory,
+			expectedInstallDirectory: defaultValues.InstallPath,
+			expectedDropDirectory:    defaultValues.DropPath,
+			expectedFingerprint:      "",
+			expectedTLS:              defaultValues.TLS != nil,
+			expectedTLSEnabled:       false,
+			expectedDisableProxy:     defaultValues.Proxy.Disable,
+			expectedTimeout:          defaultValues.Timeout,
+		},
+		{
+			input: `agent.download:
+  source_uri: "testing.uri"
+  sourceURI: "another.uri"
+`,
+			initialConfig:            DefaultConfig(),
+			expectedSourceURI:        "testing.uri",
+			expectedTargetDirectory:  defaultValues.TargetDirectory,
+			expectedInstallDirectory: defaultValues.InstallPath,
+			expectedDropDirectory:    defaultValues.DropPath,
+			expectedFingerprint:      "",
+			expectedTLS:              defaultValues.TLS != nil,
+			expectedTLSEnabled:       false,
+			expectedDisableProxy:     defaultValues.Proxy.Disable,
+			expectedTimeout:          defaultValues.Timeout,
+		},
+	}
+
+	l, _ := logger.NewWithLogpLevel("t", logp.ErrorLevel, false)
+	for _, tc := range testCases {
+		cfg := tc.initialConfig
+		reloader := NewReloader(cfg, l)
+
+		c, err := config.NewConfigFrom(tc.input)
+		require.NoError(t, err)
+
+		require.NoError(t, reloader.Reload(c))
+
+		require.Equal(t, tc.expectedSourceURI, cfg.SourceURI)
+		require.Equal(t, tc.expectedTargetDirectory, cfg.TargetDirectory)
+		require.Equal(t, tc.expectedInstallDirectory, cfg.InstallPath)
+		require.Equal(t, tc.expectedDropDirectory, cfg.DropPath)
+		require.Equal(t, tc.expectedTimeout, cfg.Timeout)
+
+		require.Equal(t, tc.expectedDisableProxy, cfg.Proxy.Disable)
+
+		if tc.expectedTLS {
+			require.NotNil(t, cfg.TLS)
+			require.Equal(t, tc.expectedTLSEnabled, *cfg.TLS.Enabled)
+			//require.Equal(t, tc.expectedFingerprint, cfg.TLS.CATrustedFingerprint)
```

Comment:
leftover ?

Reply:
no it's backport, i'm copying code, this is just something not backported yet